rskj-core/src/test/java/co/rsk/peg/federation/FederationChangeIT.java

```diff
@@ -0,0 +1,420 @@
+
+        // Since the proposed federation is commited, it should be null in storage
+        assertNull(federationStorageProvider.getPendingFederation());
+
+        // return the new federation
+        return pendingFederation.buildFederation(
+            Instant.EPOCH,
+            0,
+            BRIDGE_CONSTANTS.getFederationConstants(),
+            activations);
+    }
+
+    private void commitProposedFederation(ActivationConfig.ForBlock activations) {
+        // Verify that the proposed federation exists in storage
+        var proposedFederation = 
+            federationStorageProvider.getProposedFederation(BRIDGE_CONSTANTS.getFederationConstants(), activations);
+        assertTrue(proposedFederation.isPresent());
+
+        // As in commitPendingFederation util method, to avoid the SVP process
+        // we will commit directly
+        federationSupport.commitProposedFederation();
+    
+        // Since the proposed federation is commited, it should be null in storage
+        proposedFederation = 
+            federationStorageProvider.getProposedFederation(BRIDGE_CONSTANTS.getFederationConstants(), activations);
+        assertTrue(proposedFederation.isEmpty());
+    }
+    
+    private void activateNewFederation(ActivationConfig.ForBlock activations) {
+        // Move the required blocks ahead for the new powpeg to become active
+        var blockNumber = 
+            initialBlock.getNumber() + BRIDGE_CONSTANTS.getFederationConstants().getFederationActivationAge(activations);
+        Block activationBlock = mock(Block.class);
+        when(activationBlock.getNumber()).thenReturn(blockNumber);
+
+        // Now the new bridgeSupport points to the new block where the new federation
+        // is considered to be active
+        bridgeSupport = getBridgeSupportFromExecutionBlock(activationBlock, activations);
+    }
+    
+    private BridgeSupport getBridgeSupportFromExecutionBlock(Block executionBlock, ActivationConfig.ForBlock activations) {
+        FederationSupport fedSupport = FederationSupportBuilder.builder()
+            .withFederationConstants(BRIDGE_CONSTANTS.getFederationConstants())
+            .withFederationStorageProvider(federationStorageProvider)
+            .withRskExecutionBlock(executionBlock)
+            .build();
+
+        return BridgeSupportBuilder.builder()
+            .withProvider(bridgeStorageProvider)
+            .withRepository(repository)
+            .withEventLogger(bridgeEventLogger)
+            .withExecutionBlock(executionBlock)
+            .withActivations(activations)
+            .withBridgeConstants(BRIDGE_CONSTANTS)
+            .withBtcBlockStoreFactory(btcBlockStoreFactory)
+            .withPeginInstructionsProvider(new PeginInstructionsProvider())
+            .withFederationSupport(fedSupport)
+            .withFeePerKbSupport(feePerKbSupport)
+            .withLockingCapSupport(lockingCapSupport)
+            .build();
```

Comment:
```suggestion
        bridgeSupport = BridgeSupportBuilder.builder()
            .withProvider(bridgeStorageProvider)
            .withRepository(repository)
            .withEventLogger(bridgeEventLogger)
            .withExecutionBlock(executionBlock)
            .withActivations(activations)
            .withBridgeConstants(BRIDGE_CONSTANTS)
            .withBtcBlockStoreFactory(btcBlockStoreFactory)
            .withPeginInstructionsProvider(new PeginInstructionsProvider())
            .withFederationSupport(fedSupport)
            .withFeePerKbSupport(feePerKbSupport)
            .withLockingCapSupport(lockingCapSupport)
            .build();
```

no big deal, just to avoid the method to return

Reply:
This is set as a getter so we can expect it to return the bridgeSupport. This is how the legacy tests simulates an advance in the blockchain.